pennylane/optimize/qnspsa.py

```diff
@@ -409,13 +407,13 @@ def _get_overlap_tape(self, cost, args1, args2, kwargs):
         op_forward = self._get_operations(cost, args1, kwargs)
         op_inv = self._get_operations(cost, args2, kwargs)
 
-        with qml.tape.QuantumTape() as tape:
+        with qml.queuing.AnnotatedQueue() as q:
             for op in op_forward:
                 qml.apply(op)
             for op in reversed(op_inv):
                 op.adjoint()
             qml.probs(wires=cost.tape.wires.labels)
-        return tape
+        return qml.tape.QuantumScript.from_queue(q)
```

Comment:
This could probably just be accomplished by:
```
new_ops = op_forward + [op.adjoint() for op in op_inv]
return qml.tape.QuantumScript(new_ops, [qml.probs(wires=cost.tape.wires.labels)])
```
But not that big of deal.

Reply:
adding now